walletdb/bdb/db.go

```diff
@@ -375,6 +387,30 @@ func openDB(dbPath string, create bool, options *bbolt.Options) (walletdb.DB, er
 	// Specify bbolt freelist options to reduce heap pressure in case the
 	// freelist grows to be very large.
 	options.FreelistType = bbolt.FreelistMapType
+
+	if !create {
+		// The other value that we want to set is the initial memory
+		// map size.  When bolt expands the mmap, it actually copies
+		// over the entire database. By setting this to a large value
+		// than zero, then we can avoid some of that heap pressure due
+		// to the copies.
+		dbFileInfo, err := os.Stat(dbPath)
+		if err != nil {
+			return nil, err
+		}
+
+		// We'll aim to set the initial memory map to 2x the size of
+		// the database as it exists on disk, meaning the DB size can
+		// double without us having to remap everything.
+		mmapSize := dbFileInfo.Size() * 2
+
+		// If we're on a 32-bit system, then we'll need to limit this
+		// value to ensure we don't run into errors down the line.
+		if !is64Bit {
+			mmapSize = max32BitMapSize
```

Comment:
If we always set this for 32-bit systems, this will fail bc the process may be using other virtual memory. Also the mmap is a contiguous block of vmem, so even if this value is lower, this problem can still occur. Part of the reason why I think 32-bit support should be dropped